server/src/main/java/org/elasticsearch/index/mapper/flattened/FlattenedFieldMapper.java

```diff
@@ -163,17 +163,19 @@ public FlattenedFieldMapper build(ContentPath contentPath) {
      */
     public static final class KeyedFlattenedFieldType extends StringFieldType {
         private final String key;
+        private final String rootName;
 
-        public KeyedFlattenedFieldType(String name, boolean indexed, boolean hasDocValues, String key,
+        public KeyedFlattenedFieldType(String name, String rootName, boolean indexed, boolean hasDocValues, String key,
```

Comment:
I think `rootName` will always be the same as `name` here?

Reply:
Unfortunately not. We add the "._keyed" prefix to the name. In an earlier version I removed that from `name()` in the SourceValueFetcher but thought this was clearer to understand later.

Reply:
I wonder if we could simplify the signature by only passing in `rootName`, and then internally constructing `name` from it as `rootName + KEYED_FIELD_SUFFIX`?